flink-kubernetes/src/test/java/org/apache/flink/kubernetes/KubernetesResourceManagerTest.java

```diff
@@ -323,6 +330,18 @@ public void testGetCpuCoresNumSlots() {
 		assertThat(resourceManager.getCpuCores(configuration), is(3.0));
 	}
 
+	@Test
+	public void testCreateTaskManagerPodFailedAndRetry() throws Exception {
+		flinkKubeClient = spy(createFlinkKubeClient());
+		TestingKubernetesResourceManager testRM = createAndStartResourceManager(flinkConfig);
```

Comment:
Not sure, how this would work, `KubernetesResourceManager` creates its own `kubeClient` internally, different from `flinkKubeClient`. I would suggest to inject `kubeClient` into the constructor of `KubernetesResourceManager` in `KubernetesResourceManagerFactory`.

Reply:
This could work because we override the `createFlinkKubeClient` in `TestingKubernetesResourceManager`. Injecting the `kubeClient` into the constructor of `KubernetesResourceManager` also makes sense to me. I will go in this way.